src/Utilities/Compiler/WellKnownTypeProvider.cs

```diff
@@ -126,5 +205,45 @@ internal bool IsTaskOfType([NotNullWhen(returnValue: true)] ITypeSymbol? typeSym
                 && namedTypeSymbol.TypeArguments.Length == 1
                 && typeArgumentPredicate(namedTypeSymbol.TypeArguments[0]);
         }
+
+        private static (string[] NamespaceNames, string SimpleTypeName) GetSimpleNameInfoFromFullTypeName(string fullTypeName)
```

Comment:
I'm sure there are weird cases I'm not aware of for this.

Reply:
@mavasani @sharwell I'll change this to only try to get the namespace names, but know of any good unit tests in the compiler for parsing out namespace names from a fully qualified type name?

Reply:
I think what I have now is good enough.